build.gradle

```
@@ -582,6 +582,154 @@ task mapNamedJar(dependsOn: ["mergeV2", mapIntermediaryJar]) {
 	}
 }
 
+import daomephsta.unpick.api.ConstantUninliner
+import daomephsta.unpick.api.constantmappers.ConstantMappers
+import daomephsta.unpick.api.constantresolvers.ConstantResolvers
+import daomephsta.unpick.api.IClassResolver
+import org.objectweb.asm.ClassReader
+import java.util.jar.JarEntry
+import java.util.jar.JarFile
+import java.util.jar.JarOutputStream
+import org.objectweb.asm.tree.ClassNode
+import org.objectweb.asm.ClassWriter
+
+class JarClassResolver implements IClassResolver, Closeable {
+	private final URLClassLoader classLoader;
+
+	JarClassResolver(Set<File> jars) {
+		this.classLoader = new URLClassLoader(
+			jars.stream().map{file -> file.toURI().toURL()}.toArray{new URL[it]}
+		)
+	}
+
+	JarClassResolver(File jar) {
+		this(Collections.singleton(jar))
+	}
+
+	@Override
+	ClassReader resolveClass(String internalName) throws IOException {
+		def resource = classLoader.getResourceAsStream(internalName + ".class")
+
+		if (resource != null) {
+			return new ClassReader(resource)
+		}
+
+		throw new RuntimeException("Failed to find ${internalName}.class")
```

```suggestion
		throw new IOException("Failed to find ${internalName}.class")
```
Might be a bit more appropriate here, and it's a part of the method signature as well.